### PR TITLE
Fixing an issue with async function parsing in undo defer mode

### PR DESCRIPTION
**Pull request by aneeshdk**
In undo defer mode, for anonymous function we were not skipping the async
keyword before the function keyword.

### CODE REVIEW

**Comment by aneeshdk** [comment]
@akroshg @tcare @pleath Can you please take a look?

File: test/es7/rlexe.xml [hunk 1]
```diff
@@ -28,6 +28,13 @@
   </test>
   <test>
     <default>
+      <files>asyncawait-undodefer.js</files>
```
Comment (tcare) [inline test/es7/rlexe.xml]: The normal async await file doesn't cover these cases with -ForceUndoDefer?

**Comment by akroshg** [comment]
:shipit:

File: test/es7/asyncawait-undodefer.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+async function asyncFunc1(a) {
+    return a * a;
+}
+
+asyncFunc1(10).then(
+    result => print(result),
```
Comment (aneeshdk) [inline test/es7/asyncawait-undodefer.js]: Due to the async nature I have to create a baseline. Also printing all the results to the baseline helps in avoiding the need for counting the passing checks in the test file.

**Comment by aneeshdk** [comment]
@dotnet-bot test   OSX osx_osx_debug_static  please

**Pull request merged**